lib/nonblocking-csv-stringify.js

```diff
@@ -30,28 +30,33 @@ function nonblockingStringify(data, callback, options) {
     }
     loop();
   });
-  let chunks = [];
+
   stringifier.on('readable', function () {
     let chunk;
     while ((chunk = stringifier.read()) !== null) {
-      chunks.push(chunk);
+      callback(null, chunk);
```

Comment:
Rather than relying on the consumer needing to use a callback, did you consider relying on the fact that `stringifier` follows the standard Node Stream API? That, is if this function returned `stringifier` itself, we could pipe it directly to an Express response:

```js
// In an Express handler:
const stringifier = nonblockingStringify(data, options);
stringifier.pipe(res)
```

In fact, at that point, `nonblockingStringify` doesn't actually do anything useful, so we can just use `csv-stringify` directly:

```js
const { Stringifier } = require('csv-stringify');

app.get('/', (res) => {
  const data = getDataSomehow();
  const stringifier = new Stringifier(data);
  stringifier.pipe(res);
});